### PR TITLE
Use latest PMAT with new pick_markets strategy

### PR DESCRIPTION
**Pull request by kongzii**
## Summary by CodeRabbit

- **New Features**
  - Introduced a feature that checks for recent betting activity to optimize market selection.
  - Set a limit of participating in a maximum of 5 markets per run to enhance performance.

- **Enhancements**
  - Improved market picking logic to skip markets recently betted on, ensuring a diverse range of market engagements.

### CODE REVIEW

**Comment by kongzii** [comment]
@evangriffiths DeployableCoinFlipAgent and DeployableKnownOutcomeAgent have their custom `pick_markets` logic. Seemed like they shouldn't be updated to this, is that right?

File: prediction_market_agent/agents/think_thoroughly_agent/deploy.py [hunk 1]
```diff
@@ -1,84 +1,19 @@
...
 from prediction_market_agent.agents.think_thoroughly_agent.think_thoroughly_agent import (
     CrewAIAgentSubquestions,
 )
-from prediction_market_agent.utils import APIKeys
 
 
 class DeployableThinkThoroughlyAgent(DeployableAgent):
     model: str = "gpt-4-turbo-2024-04-09"
-    max_markets_per_run = 5
+    bet_on_n_markets_per_run = 1
```
Comment (evangriffiths) [inline prediction_market_agent/agents/think_thoroughly_agent/deploy.py:12]: to make it cheaper?

Comment (kongzii) [reply prediction_market_agent/agents/think_thoroughly_agent/deploy.py:12]: yeah, so far, he hasn't proved himself to be so good to be so expensive 😄 or do you think we should just keep it?

Comment (evangriffiths) [reply prediction_market_agent/agents/think_thoroughly_agent/deploy.py:12]: No I think that's okay - our openai bill is getting out of hand a bit, so I'll do the same for the mech agents

**Pull request merged**